### PR TITLE
pkg/security(ticdc): support online load tls config (#7927)

### LINKED ISSUE #7908: Support online reload tls in ticdc
### Is your feature request related to a problem?

Usually, there will be an expiration of the TLS certificates, we have to support online reload with new certificates instead of just restart the process.

### Describe the feature you'd like

no

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #7908

### What is changed and how it works?
1. use GetClientCertificate and GetCertificate in tls.config

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`support online load tls config in TiCDC`.
```

### CODE REVIEW

File: pkg/security/credential.go [hunk 1]
```diff
@@ -17,10 +17,11 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"encoding/pem"
+	"io/ioutil"
```
Comment (3AceShowHand) [inline pkg/security/credential.go:20]: `ioutil` looks already deprecated.

**Review by 3AceShowHand — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**